hs_composite_resource/models.py

```diff
@@ -86,7 +146,15 @@ def get_folder_aggregation_object(self, dir_path):
         """
 
         aggregation_path = dir_path[len(self.file_path) + 1:]
-        return self.filesetlogicalfile_set.filter(folder=aggregation_path).first()
+        # first check for model program/instance aggregation
+        mp_mi_aggr = self.modelprogramlogicalfile_set.filter(folder=aggregation_path).first()
```

Comment:
Can't you just filter abstractlogicalfile with the folder and the resource and return that instead of running the same filter on three different sets.

Reply:
Any abstract model class does not have a table representation in the database. So it is not possible to use such a model for database query.